src/gs-screenshot-image.c

```diff
@@ -368,26 +471,42 @@ gs_screenshot_image_load_async (GsScreenshotImage *ssimg,
 		sizedir = g_strdup_printf ("%ux%u", ssimg->width * ssimg->scale, ssimg->height * ssimg->scale);
 	}
 	cache_kind = g_build_filename ("screenshots", sizedir, NULL);
-	ssimg->filename = gs_utils_get_cache_filename (cache_kind,
-						       basename,
-						       GS_UTILS_CACHE_FLAG_NONE,
-						       NULL);
-	if (ssimg->filename == NULL) {
+
+	/* request the writable cache filename before trying the system's one
+	 * because the former should be the more recent */
+	writable_filename = gs_utils_get_cache_filename (cache_kind,
+							 basename,
+							 GS_UTILS_CACHE_FLAG_WRITEABLE,
+							 NULL);
+
+	if (writable_filename == NULL) {
 		/* TRANSLATORS: this is when we try create the cache directory
 		 * but we were out of space or permission was denied */
 		gs_screenshot_image_set_error (ssimg, _("Could not create cache"));
 		return;
 	}
 
-	/* does local file already exist */
-	if (g_file_test (ssimg->filename, G_FILE_TEST_EXISTS)) {
-		as_screenshot_show_image (ssimg);
-		return;
+	if (ssimg->filename != NULL) {
+		g_free (ssimg->filename);
+	}
+
+	/* if the user's cache file does not exist, try to use the system one */
+	if (!g_file_test (writable_filename, G_FILE_TEST_EXISTS)) {
+		ssimg->filename = gs_utils_get_cache_filename (cache_kind,
+							       basename,
+							       GS_UTILS_CACHE_FLAG_NONE,
+							       NULL);
+	} else {
+		ssimg->filename = g_steal_pointer (&writable_filename);
 	}
 
-	/* can we load a blurred smaller version of this straight away */
-	if (ssimg->width > AS_IMAGE_THUMBNAIL_WIDTH &&
-	    ssimg->height > AS_IMAGE_THUMBNAIL_HEIGHT) {
+	/* show the screenshot if its cached version exists, otherwise try to
+	 * load a blurred smaller version while we fetch the new image */
+	if (g_file_test (ssimg->filename, G_FILE_TEST_EXISTS)) {
```

Comment:
You probably only want to do this check when in the code path of the first case in the above if